mimium-lang/src/compiler/mirgen.rs

```diff
@@ -347,6 +344,43 @@ impl Context {
         };
         Ok(v)
     }
+    fn eval_assign(
+        &mut self,
+        name: Symbol,
+        src: VPtr,
+        t: TypeNodeId,
+        span: &Span,
+    ) -> Result<(), CompileError> {
+        match self.lookup(&name) {
+            LookupRes::Local(v) => match v.as_ref() {
+                Value::Argument(_i, _a) => Err(CompileError(
+                    CompileErrorKind::AssignmentToArg,
+                    span.clone(),
+                )),
+                _ => {
+                    self.push_inst(Instruction::Store(v.clone(), src, t));
+                    Ok(())
+                }
+            },
+            LookupRes::UpValue(_level, upv) => {
```

Comment:
I think I'm ready to merge it but it is still unclear whether deeply-nested closure and destructive assignments are treated correctly.
Level information is not used compared to the case of getupvalue.